tests/app_test.py

```diff
@@ -0,0 +1,194 @@
+from flask.wrappers import Response
+from controllers import app
+from controllers import db
+from flask_sqlalchemy import SQLAlchemy
+from services.data_service import DataService
+from datetime import datetime
+from models import *
+import json
+import os
+
+tester = app.app.test_client()
+
+QA_MODEL_CONTROLLER_VERSION = 'v1'
+
+def test_root():
+    response = tester.get("/", content_type="html/text")
+
+    assert response.status_code == 200
+    assert response.data == b'"The TweetQA Api is running!"\n'
+
+def mock_db_setup():
+    app.app.config['TESTING'] = True
+    app.app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:///test.db'
+    db.drop_all()
+    db.create_all()
+    db.session.commit()
+
+def mock_db_teardown():
+    db.session.close()
+    os.remove('controllers/test.db')
+
+def test_data_service_creates_data():
+    '''TC-001: The DataService object can create a datum entity in the database'''
+    mock_db_setup()
+
+    data_in = Data(
+        qid = 'test qid1',
+        tweet = ('test tweet'),
+        question = 'test question',
+        answer = 'test answer',
+        created_date = datetime.now(),
+        updated_date = datetime.now(),
+        source = "original dataset",
+        start_position = 1,
+        end_position = 4
+    )
+
+    data_service = DataService()
+    data_out = data_service.create_data(data_in)
+    actual_data = Data.query.filter(Data.id == data_in.id).first()
+    mock_db_teardown()
+
+    assert actual_data is not None
+    assert data_out is not None
+    assert actual_data == data_out
+    assert data_out == data_in
+
+def test_data_service_reads_data_by_id():
+    '''TC-002: The DataService object can read a datum entity from the database by id'''
+    mock_db_setup()
+
+    data_in = Data(
```

Comment:
Data object can be created using faker and cooky cutter, then you don't have to hard code it each time, you just ask for a new instance of Data.  We can create similar mock objects for all of our data types.